clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2453,6 +2453,10 @@ void CodeGenModule::SetLLVMFunctionAttributesForDefinition(const Decl *D,
       // function would give a trace where `F` was actually inlined instead of
       // outlined.
       B.addAttribute(llvm::Attribute::NoInline);
+      // Remove the `AlwaysInlined` attribute if it's set, so it doesn't clash
```

Comment:
This doesn't feel like the right place for this? I worry it'll be forgotten (and may not be the right thing for the optimised multimodule too). Can we put it in a yk pass or ...?

Reply:
Why wouldn't this be the right place? `NoInline` and `AlwaysInline` clash, so when we set `NoInline` we need to make sure we remove `AlwaysInline`. If we are setting `NoInline` in this place, we also should also remove `AlwaysInline` in the same place.

The optimised multimodule will have to do the same thing if it sets `NoInline`.

Reply:
But we are clearly setting `NoInline` here, so it needs to be paired with removing `AlwaysInline`.

The problem with a separate pass is that if a user adds `noinline` and `alwaysinline` manually in the code somewhere, our pass favour `noinline` and silently remove `alwaysinline` and the user has no idea this happened.

So I believe we should only do this in places where we actively force `noinline` in one of our passes/changes.

Reply:
Ah I see the confusion now. Yes the `NoInline` is one of our changes.